src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/ThemeManager3.cs

```diff
@@ -0,0 +1,434 @@
+
+    #endregion
+
+
+    #region Helper Methods
+
+    private static bool GetUseLightColors(ThemeMode themeMode)
+    {
+        // Is this needed ?
+        if(themeMode == ThemeMode.None)
+        {
+            return true;
+        }
+
+        // Do we need to add a check for ThemeMode.None theme?
+        return themeMode == ThemeMode.Light || (themeMode == ThemeMode.System && IsSystemThemeLight());
+    }
+
+    private static bool ResourceDictionaryContainsFluentDictionary(ResourceDictionary rd, out ThemeMode themeMode)
+    {
+        themeMode = ThemeMode.None;
+        if (rd == null) return false;
+
+        int index = FindLastFluentThemeResourceDictionaryIndex(rd);
+
+        if(index != -1)
+        {
+            string dictionarySource = rd.MergedDictionaries[index].Source.ToString();
+            if(dictionarySource.EndsWith("Fluent.Light.xaml", StringComparison.OrdinalIgnoreCase))
+            {
+                themeMode = ThemeMode.Light;
+            }
+            else if(dictionarySource.EndsWith("Fluent.Dark.xaml", StringComparison.OrdinalIgnoreCase))
+            {
+                themeMode = ThemeMode.Dark;
+            }
+            else
+            {
+                themeMode = ThemeMode.System;
+            }
+            return true;
+        }
+        return false;
+    }
+
+    private static Uri GetFluentThemeResourceUri(bool useLightMode)
+    {
+        string themeFileName = "Fluent." + (useLightMode ? "Light" : "Dark") + ".xaml";
+
+        if(SystemParameters.HighContrast)
+        {
+            themeFileName = "Fluent.HC.xaml";
+        }
+
+        return new Uri(fluentThemeResoruceDictionaryUri + themeFileName, UriKind.Absolute);
+
+    }
+
+    private static void AddOrUpdateThemeResources(ResourceDictionary rd, Uri dictionaryUri)
+    {
+        if (rd == null) return;
+
+        ArgumentNullException.ThrowIfNull(dictionaryUri, nameof(dictionaryUri));
+        
+        var newDictionary = new ResourceDictionary() { Source = dictionaryUri };
+        int index = FindLastFluentThemeResourceDictionaryIndex(rd);
+
+        if (index >= 0)
+        {
+            rd.MergedDictionaries[index] = newDictionary;
+        }
+        else
+        {
+            rd.MergedDictionaries.Insert(0, newDictionary);
+        }
+    }
+
+    private static int FindLastFluentThemeResourceDictionaryIndex(ResourceDictionary rd)
+    {
+        // Return -1 or throw ?
+        // Throwing here because, here we are passing application or window resources,
+        // and even though when the field is null, a new RD is created and returned.
+        ArgumentNullException.ThrowIfNull(rd, nameof(rd));
+
+        for(int i = rd.MergedDictionaries.Count - 1; i >= 0; i--)
+        {
+            if(rd.MergedDictionaries[i].Source != null)
+            {
+                if(rd.MergedDictionaries[i].Source.ToString().StartsWith(fluentThemeResoruceDictionaryUri))
```

Comment:
Should we add the StringComparison here?

```suggestion
                if(rd.MergedDictionaries[i].Source.ToString().StartsWith(fluentThemeResoruceDictionaryUri, StringComparison.Ordinal))
```